src/suggestion.cpp

```diff
@@ -0,0 +1,325 @@
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
+ *
+ */
+
+#define ZIM_PRIVATE
+
+#include <zim/suggestion.h>
+#include <zim/item.h>
+#include "suggestion_internal.h"
+#include <iostream>
+#include "fileimpl.h"
+#include "tools.h"
+#include "constants.h"
+
+#include <unicode/locid.h>
+
+namespace zim
+{
+
+SuggestionDataBase::SuggestionDataBase(const Archive& archive)
+  : m_archive(archive)
+{
+  m_queryParser.set_database(m_database);
+  m_queryParser.set_default_op(Xapian::Query::op::OP_AND);
+
+  auto impl = archive.getImpl();
+  FileImpl::FindxResult r;
+
+  r = impl->findx('X', "title/xapian");
+  if (!r.first) {
+    return;
+  }
+
+  auto xapianEntry = Entry(impl, entry_index_type(r.second));
+  auto accessInfo = xapianEntry.getItem().getDirectAccessInformation();
+  if (accessInfo.second == 0) {
+      return;
+  }
+
+  DEFAULTFS::FD databasefd;
+  try {
+      databasefd = DEFAULTFS::openFile(accessInfo.first);
+  } catch (...) {
+      std::cerr << "Impossible to open " << accessInfo.first << std::endl;
+      std::cerr << strerror(errno) << std::endl;
+      return;
+  }
+  if (!databasefd.seek(offset_t(accessInfo.second))) {
+      std::cerr << "Something went wrong seeking databasedb "
+                << accessInfo.first << std::endl;
+      std::cerr << "dbOffest = " << accessInfo.second << std::endl;
+      return;
+  }
+
+  Xapian::Database database;
+  try {
+      database = Xapian::Database(databasefd.release());
+  } catch( Xapian::DatabaseError& e) {
+      std::cerr << "Something went wrong opening xapian database for zimfile "
+                << accessInfo.first << std::endl;
+      std::cerr << "dbOffest = " << accessInfo.second << std::endl;
+      std::cerr << "error = " << e.get_msg() << std::endl;
+      return;
+  }
+
+  m_valuesmap = read_valuesmap(database.get_metadata("valuesmap"));
+  auto language = database.get_metadata("language");
+  if (language.empty() ) {
+      // Database created before 2017/03 has no language metadata.
+      // However, term were stemmed anyway and we need to stem our
+      // search query the same the database was created.
+      // So we need a language, let's use the one of the zim.
+      // If zimfile has no language metadata, we can't do lot more here :/
+      try {
+          language = archive.getMetadata("Language");
+      } catch(...) {}
+  }
+  if (!language.empty()) {
+      icu::Locale languageLocale(language.c_str());
+      /* Configuring language base steemming */
+      try {
+          m_stemmer = Xapian::Stem(languageLocale.getLanguage());
+          m_queryParser.set_stemmer(m_stemmer);
+          m_queryParser.set_stemming_strategy(Xapian::QueryParser::STEM_SOME);
+      } catch (...) {
+          std::cout << "No stemming for language '" << languageLocale.getLanguage() << "'" << std::endl;
+      }
+  }
+
+  m_database.add_database(database);
```

Comment:
We added the database to `m_database` in search to support multi search. We can simply set `m_database` here.